lib/components/Input/base/index.tsx

```diff
@@ -171,30 +196,31 @@ const Input = forwardRef<HTMLInputElement, InputProps>(
                                         {leftPlaceholder}
                                     </Text>
                                 )}
-                            <input
-                                {...rest}
-                                readOnly={readOnly}
-                                required={required}
-                                type={type}
-                                value={inputValue}
-                                maxLength={maxLength}
-                                placeholder={placeholder}
-                                className={clsx(
-                                    "input",
-                                    "peer",
-                                    `input__align--${textAlignment}`,
-                                    `input__size--${inputSize}`,
-                                )}
-                                disabled={!!disabled}
-                                onChange={(e) => {
-                                    setInputValue(e.target.value);
-                                    onChange?.(e);
-                                }}
-                                onFocus={() => setFocused(true)}
-                                onBlur={() => setFocused(false)}
-                                id={label?.toString()}
-                                ref={ref}
-                            />
+                            {formatProps ? (
+                                <PatternFormat
```

Comment:
When consumers need to use masking then they can pass in formatProps to use the PatternFormat component.